File: typescript/random-terraform-stack.ts
import { Construct } from "constructs";
import { Integer, Password, RandomProvider } from "./.gen/providers/random"
import { TerraformStack } from "cdktf";

export class RandomTerraformStack extends TerraformStack {
  constructor(scope: Construct, name: string) {
    super(scope, 
      name);

    new RandomProvider(this, "random");

    var len = new Integer(this, "integer", {
      min: 0,
      max: 10
    });

    new Password(this, "password", {
      length: len.result
    });
  }
}

File: python/RandomTerraformStack.py
#!/usr/bin/env python
from os import environ

from cdktf import TerraformStack
from constructs import Construct

from imports.random import Integer, Password, RandomProvider

class RandomTerraformStack(TerraformStack):
    def __init__(self, scope: Construct, ns: str):
        super().__init__(scope, ns)

        RandomProvider(self, 'random')
    
        length = Integer(self, "length", min=5, max=10)
        
        Password(self, f"password",
          length=length.result
        )
